dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -235,30 +251,80 @@ class OpenDDS_Rtps_Udp_Export RtpsUdpDataLink : public DataLink {
 
   typedef OPENDDS_MAP_CMP(RepoId, ReaderInfo, GUID_tKeyLessThan) ReaderInfoMap;
 
-  struct RtpsWriter {
+  class  RtpsWriter : public RcObject {
+  protected:
```

Comment:
I didn't want RtpsUdpDataLink's methods tinkering around with the internals of RtpsWriter, which this seemed to prevent. If you'd rather this be private, I'm OK with that for now, although someday it may make sense to split RtpsWriter into reliable and non-reliable subclasses so we can move the handling of non-reliable stuff out of RtpsUdpDataLink's methods. With that in mind, I don't see any harm in leaving them protected.